src/Native/Runtime/thread.cpp

```diff
@@ -1118,7 +1121,16 @@ FORCEINLINE bool Thread::InlineTryFastReversePInvoke(ReversePInvokeFrame * pFram
 void Thread::ReversePInvokeAttachOrTrapThread(ReversePInvokeFrame * pFrame)
 {
     if (!IsStateSet(TSF_Attached))
+    {
+        if (*InitializeRuntimePtr != NULL && !INITIALIZING_RUNTIME)
+        {
+            INITIALIZING_RUNTIME = true;
+            (*InitializeRuntimePtr)();
+            INITIALIZING_RUNTIME = false;
```

Comment:
@jkotas I initially had this in the runtime like you advised but I noticed that InitializeRuntime calls InitializeModules which in turn calls this method and leads to a stack overflow. Any recommentdations?

Reply:
I think that the Thread* of the thread initializing the runtime should be store in a static. If you are coming back here on the thread that is initializing the runtime already, we know that the call is part of the initialization - skip the check in that case.